packages/react/src/hooks/useModal.ts

```diff
@@ -1,43 +1,36 @@
-import { useContext, useEffect } from 'react'
-
-import { FrigadeContext } from '@/components/Provider'
+import { useCallback, useEffect } from 'react'
 import type { Flow } from '@frigade/js'
 
+export let globalModalState: Set<string> = new Set()
+
 export function useModal(flow: Flow, isModal: boolean = true) {
-  const { currentModal, modals, setModals } = useContext(FrigadeContext)
+  const removeModal = useCallback(() => {
+    if (globalModalState.has(flow?.id)) {
+      globalModalState.delete(flow?.id)
+    }
+  }, [globalModalState, isModal])
 
-  useEffect(() => {
-    if (isModal && flow?.isVisible && flow && !modals.has(flow.id)) {
-      setModals((prevModals) => new Set(prevModals).add(flow.id))
+  const registerModal = useCallback(() => {
+    if (isModal && flow?.isVisible && flow && !globalModalState.has(flow.id)) {
```

Comment:
Can remove `&& flow`, flow must exist in order for `flow.visible` to be true.

Reply:
Good catch